File: 卡码/代码随想录/Go/二叉树/02-二叉树属性/LeetCode_0113.go
//113. 路径总和 II
package main

import (
	. "binaryTree/types"
)

var (
	res [][]int
	path []int
)

func traversal(root *TreeNode,targetSum int){
	if root==nil {
		return
	}
	path=append(path,root.Val)
	//叶子节点且sum等于当前节点值说明找到路径
	if root.Left==nil&&root.Right==nil&&targetSum==root.Val {
		tmp:=make([]int,len(path))
		copy(tmp,path)
		res=append(res,tmp)
	}
	traversal(root.Left,targetSum-root.Val)
	traversal(root.Right,targetSum-root.Val)
	path=path[:len(path)-1]
}

func pathSum(root *TreeNode, targetSum int) [][]int {
	res=make([][]int,0)
	path=make([]int,0)
	traversal(root,targetSum)
	return res
}

File: 卡码/代码随想录/JS/二叉树/02-二叉树属性/LeetCode_0113.js
//113. 路径总和 II
import TreeNode from "../TreeNode.js";

let res=new Array();
let path=new Array();
const traversal=function(root,sum){
  if(!root) return;
  path.push(root.val);
  //叶子节点且sum等于当前节点值说明找到路径
  if(!root.left&&!root.right&&sum==root.val) res.push([...path]);
  traversal(root.left,sum-root.val);
  traversal(root.right,sum-root.val);
  path.pop();
}
const pathSum = function(root, targetSum) {
  res=[];
  path=[];
  traversal(root,targetSum);
  return res;
};